src/fixedpoint.ts

```diff
@@ -1,7 +1,21 @@
+/**
+ * Fixed point number implementation.
+ *
+ * This is useful for representing token amounts and other big numbers.
+ */
 export class FixedInt<T extends number> {
   _value: bigint;
+  /**
+   * The number of decimals to use.
+   */
   decimals: T;
 
+  /**
+   * Create a new fixed point number.
+   *
+   * @param val The value of the fixed point number.
+   * @param decimals The number of decimals to use.
+   */
   constructor(val: bigint, decimals: T) {
     if (decimals < 0) {
       throw new Error("decimals must be greater than or equal to 0");
```

Comment:
Suggestion: Lets define a lib error to return, something like FPNumError, so its simpler to delineate between another unexpected `Error` and this one. Will be useful when writing code that is related to user input that leans on exception handling

Reply:
GREAT great idea. lfg